src/supa/connection/provider/server.py

```diff
@@ -197,9 +244,47 @@ def ReserveAbort(
         log = logger.bind(method="ReserveAbort")
         logger.info("Received message.", request_message=pb_reserve_abort_request)
 
-        from supa import scheduler
+        connection_id = UUID(pb_reserve_abort_request.connection_id)
+
+        from supa.db.session import db_session
 
-        scheduler.add_job(ReserveAbortJob(UUID(pb_reserve_abort_request.connection_id)))
+        with db_session() as session:
```

Comment:
Haven't looked into detail (yet), but there seems to be a lot of duplicate code on the next lines

Reply:
That is correct, the only difference is the type of request handled, the state transition, the job started and the response returned. But the structure of the code is more or less the same. I could parameterise the code but doubt if that makes it more readable.